### DISCUSSION TITLE
understanding the source code

### DISCUSSION THREAD

**Discussion by kshitij86** [mitmproxy/mitmproxy#5473] [Q&A] [answered]
as a beginner what should be the "flow" (pun intended) to understand the source code? what should be a starting point to better understand the core functionalities of `mitmproxy`. as the codebase is huge I find myself a bit lost so a few tips are greatly appreciated so I can start contributing.

**Comment by mhils** [comment, marked as answer]
I think the main strategy here is not trying to understand everything at once, but rather focus on individual subsystems. If you have specific questions on the code, I'd encourage you to check out our developer Slack. :)
